lib/network/net.go

```diff
@@ -185,15 +200,24 @@ func (c *TcpConn) Send(ctx context.Context, obj ProtocolMessage) error {
 	}
 	//c.Conn.SetWriteDeadline(time.Now().Add(timeOut))
 	// First write the size
-	packetSize := Size(len(b))
-	if err := binary.Write(c.Conn, globalOrder, packetSize); err != nil {
+	packetSize := uint64(len(b))
+
+	// FIXME Write the size as an UvarInt
+	var sb = make([]byte, binary.MaxVarintLen64)
+	n := binary.PutUvarint(sb[:], packetSize)
```

Comment:
I did `git checkout sda_1 -- net.go`  and the same with struct.go and encoding.go to see if messed up sth: As expected it does not block but I now get a `networfatal error: bad map state` [sic] and `invalid memory address or nil pointer dereference` and much more. So at least our changes didn't make it worse...